service/controller/resource/monitoring/prometheus/test/case-5-eks-v18.golden

```diff
@@ -38,7 +38,7 @@ spec:
     customer: Giant Swarm
     installation: test-installation
     pipeline: testing
-    provider: eks
+    provider: aws
```

Comment:
The tests look odd.
For instance, we have some file named `./service/controller/resource/monitoring/scrapeconfigs/test/openstack/case-2-aws-v16.golden`.
It used to set `replacement: openstack`, but now sets `replacement: capa`.
I don't really understand how these tests work, but are they supposed to simulate a `openstack` config, a vintage `aws` config, on a `capa` config?

Reply:
There is a big issue with those tests now :( 

First, openstack should be removed but we now set replacement: capa because the cluster CR reference an AWSCluster CR. 

For the eks case, the issue is that our test actually renders an EKS (AWSManagedCluster) cluster on a Vintage MC so the condition is wrong from the start ...

Reply:
I think the main issue is that the tests are generally Vintage focused more than CAPI focused :(

Reply:
For this use case, we go through this case https://github.com/giantswarm/prometheus-meta-operator/blob/92f9a43b250ca401efbe03e5978bbeee453c0006/service/key/key.go#L124 (IsVintageManagementCluster because of this https://github.com/giantswarm/prometheus-meta-operator/blob/92f9a43b250ca401efbe03e5978bbeee453c0006/service/controller/resource/monitoring/prometheus/resource_test.go#L81)

Reply:
I was mostly concerned that the "broken" tests show broken logic in the code.

If we're sure that the problem comes from the test, we could probably merge now without fixing the tests. But then, I'm not sure we can spot future mistakes with the tests.
So, I think we should fix the tests. But if it's in a separate PR that looks fine to me.